src/lib.rs

```diff
@@ -208,6 +210,62 @@ impl GetTransactionResponse {
     }
 }
 
+#[derive(serde::Deserialize, serde::Serialize, Debug, Clone)]
+pub struct GetTransactionsResponseRaw {
+    pub transactions: Vec<GetTransactionResponseRaw>,
+    #[serde(rename = "latestLedger")]
+    pub latest_ledger: u32,
+    #[serde(rename = "latestLedgerCloseTimestamp")]
+    pub latest_ledger_close_time: i64,
+    #[serde(rename = "oldestLedger")]
+    pub oldest_ledger: u32,
+    #[serde(rename = "oldestLedgerCloseTimestamp")]
+    pub oldest_ledger_close_time: i64,
+    pub cursor: String,
+}
+#[derive(serde::Deserialize, serde::Serialize, Debug, Clone)]
+pub struct GetTransactionsResponse {
+    pub transactions: Vec<GetTransactionResponse>,
+    pub latest_ledger: u32,
+    pub latest_ledger_close_time: i64,
+    pub oldest_ledger: u32,
+    pub oldest_ledger_close_time: i64,
+    pub cursor: String,
+}
+impl TryInto<GetTransactionsResponse> for GetTransactionsResponseRaw {
+    type Error = xdr::Error; // assuming xdr::Error or any other error type that you use
+
+    fn try_into(self) -> Result<GetTransactionsResponse, Self::Error> {
+        let mut transactions = Vec::new();
+
+        for tx_raw in self.transactions {
+            let tx: GetTransactionResponse = tx_raw.try_into()?;
+            transactions.push(tx);
+        }
+
+        Ok(GetTransactionsResponse {
+            transactions,
```

Comment:
```suggestion
        Ok(GetTransactionsResponse {
            transactions: self.transaction.into_iter().map(TryInto::try_into).collect::<Result<Vec<_>,xdr::Error>>()?,
```
Iterators are your best friend ;-). Also probably need to do this locally to do `cargo fmt`

Reply:
got it, thanks!